target/ppc/int_helper.c

```diff
@@ -910,6 +910,102 @@ VCT(uxs, cvtsduw, u32)
+    {                                                                   \
+        uint8_t pmsk, xmsk, ymsk;                                       \
+        uint8_t pmsk_bit, xmsk_bit, ymsk_bit;                           \
+        ppc_vsr_t *a, *b, *at;                                          \
+        int64_t psum;                                                   \
+        int32_t va, vb;                                                 \
+        int i, j, k;                                                    \
+        bool sat, acc;                                                  \
+        xmsk = (uint8_t)mask & 0x0F;                                    \
+        ymsk = (uint8_t)(mask >> 4) & 0x0F;                             \
+        pmsk = (uint8_t)(mask >> 8) & 0xFF;                             \
+        a = cpu_vsr_ptr(env, a_r);                                      \
+        b = cpu_vsr_ptr(env, b_r);                                      \
+        acc = (bool)(packed_flags & GER_ACC); /* Accumulate */          \
+        sat = (bool)(packed_flags & GER_SAT); /* Saturation */          \
+        for (i = 0, xmsk_bit = 1 << 3; i < 4; i++, xmsk_bit >>= 1) {    \
+            at = cpu_vsr_ptr(env, at_r + i);                            \
+            for (j = 0, ymsk_bit = 1 << 3; j < 4; j++, ymsk_bit >>= 1) {\
+                if ((xmsk_bit & xmsk) && (ymsk_bit & ymsk)) {           \
+                    psum = 0;                                           \
+                    for (k = 0, pmsk_bit = 1 << (RANK-1); k < RANK;     \
+                         k++, pmsk_bit >>= 1) {                         \
+                        if (pmsk_bit & pmsk) {                          \
+                            GER_MULT(BITS, RANK, EL);                   \
+                        }                                               \
+                    }                                                   \
+                    if (acc) {                                          \
+                        psum += at->VsrW(j);                            \
+                    }                                                   \
+                    if (sat && (psum > (1LL<<31) - 1)) {                \
+                        set_vscr_sat(env);                              \
+                        at->VsrSW(j) = (1LL<<31) - 1;                   \
+                    } else if (sat && psum < -(1LL<<31)) {              \
+                        set_vscr_sat(env);                              \
+                        at->VsrSW(j) = -(1LL<<31);                      \
+                    } else {                                            \
+                        at->VsrSW(j) = (int32_t) psum;                  \
+                    }                                                   \
+                } else {                                                \
+                    at->VsrW(j) = 0;                                    \
```

Comment:
I used ```VsrW(j)``` because the ISA defines that if not ```XMSK.bit[i] & YMSK.bit[j]``` then ```ACC[AT][i][j] ← 0x0000_0000```, so I thought that since it's attributing an hex, then unsigned seemed more fitting.

Reply:
I would use VsrSW, just to avoid confusion. 0x0000_0000 is equally 'signed 0', and both options (assigning to VsrW and to VsrSW) will perform the exact same thing.